docs/high/group.rst

```diff
@@ -158,11 +158,11 @@ link resides.
 
 .. note::
 
-    How the filename is processed is operating system dependent, it is
-    recommended to read :ref:`file_filenames` to understand potential limitations on
-    filenames on your operating system. Note especially that Windows is
-    particularly susceptible to problems with external links, due to possible
-    encoding errors and how filenames are structured.
+    The filename is stored in the file as bytes, normally UTF-8 encoded.
+    In most cases, this should work reliably, but problems are possible if a
+    file created on one platform is accessed on another. Older versions of HDF5
```

Comment:
I think cross-platform issues are the only way this could show up now - e.g. if you make an external link on Linux using a non-UTF-8 locale, then copy the files to a Windows machine in a way that correctly converts the filenames to Unicode, HDF5 will try to interpret the name from the link as UTF-8.